src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -82,20 +96,92 @@ public CreateWorkflowTransportAction(
         FlowFrameworkIndicesHandler flowFrameworkIndicesHandler,
         FlowFrameworkSettings flowFrameworkSettings,
         Client client,
-        PluginsService pluginsService
+        PluginsService pluginsService,
+        ClusterService clusterService,
+        NamedXContentRegistry xContentRegistry,
+        Settings settings
     ) {
         super(CreateWorkflowAction.NAME, transportService, actionFilters, WorkflowRequest::new);
         this.workflowProcessSorter = workflowProcessSorter;
         this.flowFrameworkIndicesHandler = flowFrameworkIndicesHandler;
         this.flowFrameworkSettings = flowFrameworkSettings;
         this.client = client;
         this.pluginsService = pluginsService;
+        filterByEnabled = FILTER_BY_BACKEND_ROLES.get(settings);
+        this.clusterService = clusterService;
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        this.xContentRegistry = xContentRegistry;
+        this.settings = settings;
     }
 
     @Override
     protected void doExecute(Task task, WorkflowRequest request, ActionListener<WorkflowResponse> listener) {
-
         User user = getUserContext(client);
+        String workflowId = request.getWorkflowId();
+        try {
+            resolveUserAndExecute(user, workflowId, listener, () -> createExecute(request, user, listener));
+        } catch (Exception e) {
+            logger.error("Failed to create workflow", e);
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Resolve user and execute the workflow function
+     * @param requestedUser the user making the request
+     * @param workflowId the workflow id
+     * @param listener the action listener
+     * @param function the workflow function to execute
+     */
+    private void resolveUserAndExecute(
+        User requestedUser,
+        String workflowId,
+        ActionListener<WorkflowResponse> listener,
+        WorkflowFunction function
+    ) {
+        try {
+            // Check if user has backend roles
+            // When filter by is enabled, block users creating/updating workflows who do not have backend roles.
+            if (filterByEnabled) {
+                String error = checkFilterByBackendRoles(requestedUser);
```

Comment:
Doesn't look like tests are covering the `filterByEnabled` case.

Reply:
We are now in the latest commit )